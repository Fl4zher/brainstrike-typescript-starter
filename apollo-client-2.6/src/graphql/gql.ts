import gql from 'graphql-tag';

export const CARD_DATA = gql`
  fragment CardData on Card {
    __typename
    created
    updated
    id
    label
    number
  }
`;

export const GET_CARD_DATA = gql`
  query getCards(
    $first: Int
    $last: Int
    $after: String
    $before: String
    $orderByColumn: String
    $orderByDirection: DirectionEnum
    $categoryId: ID
  ) {
    cards(
      first: $first
      last: $last
      after: $after
      before: $before
      orderByColumn: $orderByColumn
      orderByDirection: $orderByDirection
      categoryId: $categoryId
    ) @connection(key: "Card_card") {
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
        totalCount
      }
      edges {
        cursor
        node {
          ...CardData
        }
      }
    }
  }
  ${CARD_DATA}
`;

export const GET_CARD_CATEGORIES = gql`
  query getCategories {
    categories {
      id
      name
      parentId
      created
      updated
    }
  }
`;
