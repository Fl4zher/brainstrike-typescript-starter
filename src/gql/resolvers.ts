import { merge } from "lodash";
import { GraphQLDate, GraphQLTime, GraphQLDateTime } from "graphql-iso-date";

import cardResolvers from "./card/resolvers";
import userResolvers from "./user/resolvers";

// Default Resolvers
const resolvers = {
  Date: GraphQLDate,
  Time: GraphQLTime,
  DateTime: GraphQLDateTime
};

export default merge(resolvers, cardResolvers, userResolvers);
