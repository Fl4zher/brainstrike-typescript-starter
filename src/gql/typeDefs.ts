import { importSchema } from "graphql-import";

export const typeDefs = importSchema(
  "src/gql/schema.graphql"
); /* Warning: Must be an absolute path */

export default typeDefs;
