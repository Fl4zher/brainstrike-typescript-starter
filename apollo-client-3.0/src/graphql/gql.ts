import gql from 'graphql-tag';

export const NODE_FRAGMENT = gql`
  fragment NodeFragment on Node {
    created
    updated
  }
`;

export const CARD_FRAGMENT = gql`
  fragment CardFragment on Card {
    id
    ...NodeFragment
    __typename
    label
    number
  }
  ${NODE_FRAGMENT}
`;

export const GET_CARD_DATA = gql`
  query getCards(
    $first: Int
    $last: Int
    $after: String
    $before: String
    $orderByColumn: String
    $orderByDirection: DirectionEnum
    $categoryId: ID
  ) {
    cards(
      first: $first
      last: $last
      after: $after
      before: $before
      orderByColumn: $orderByColumn
      orderByDirection: $orderByDirection
      categoryId: $categoryId
    ) @connection(key: "Cards") {
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
        totalCount
      }
      edges {
        cursor
        node {
          ...CardFragment
        }
      }
    }
  }
  ${CARD_FRAGMENT}
`;

export const CATEGORY_FRAGMENT = gql`
  fragment CategoryFragment on Category {
    __typename
    id
    created
    updated
    name
    parentId
  }
`;

export const GET_CARD_CATEGORIES = gql`
  query getCategories {
    categories {
      ...CategoryFragment
    }
  }
  ${CATEGORY_FRAGMENT}
`;

export const GET_CATEGORY = gql`
  query getCategory(
    $id: ID!
    $first: Int
    $last: Int
    $after: String
    $before: String
    $orderByColumn: String
    $orderByDirection: DirectionEnum
  ) {
    category: node(id: $id) {
      ... on Node {
        id
        created
        updated
      }
      ... on Category {
        __typename
        name
        parentId
        _cards(
          first: $first
          last: $last
          after: $after
          before: $before
          orderByColumn: $orderByColumn
          orderByDirection: $orderByDirection
        ) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
            totalCount
          }
          edges {
            cursor
            node {
              __typename
              id
              label
              number
              created
              updated
            }
          }
        }
      }
    }
  }
`;
