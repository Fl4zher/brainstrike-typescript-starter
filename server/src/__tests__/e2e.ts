import {
  GraphQLRequest,
  Observable,
  FetchResult,
  toPromise
} from "apollo-link";
import gql from "graphql-tag";

import {
  startTestServer,
  constructTestServer,
  createTestingConnection,
  Connection
} from "./__utils";

import * as TDATA from "./__testData";
import * as GQL from "./__queries";

// NOTE: would've like to share the GQL.GET_CARD_DATA but doesn't work for some reason, will investigate later
// limit to the first 25 seeded records
const GET_CARDS = gql`
  query getCards {
    cards(first: 25) {
      pageInfo {
        hasNextPage
        hasPreviousPage
      }
      edges {
        cursor
        node {
          __typename
          id
          number
          label
        }
      }
    }
  }
`;

describe("Server - e2e", () => {
  let connection: Connection;

  let stop: () => void,
    graphql: ({}: GraphQLRequest) => Observable<FetchResult>;

  beforeAll(async () => {
    console.log("creating test connection");
    connection = await createTestingConnection();
  });

  afterAll(() => connection.close());

  beforeEach(async () => {
    const { apolloServer } = await constructTestServer(connection);
    const testServer = await startTestServer(apolloServer);
    stop = testServer.stop;
    graphql = testServer.graphql;
  });

  afterEach(async () => {
    stop();
  });

  it("gets list of cards", async () => {
    const res = await toPromise(
      graphql({
        query: GET_CARDS
      })
    );

    expect(res).toMatchSnapshot();
  });

  it("gets a single card", async () => {
    const res = await toPromise(
      graphql({
        query: GQL.GET_CARD,
        variables: { id: TDATA.mockE2EFirstId }
      })
    );
    expect(res).toMatchSnapshot();
  });
});
