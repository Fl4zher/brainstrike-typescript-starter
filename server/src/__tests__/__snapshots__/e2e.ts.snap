// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e gets a single card 1`] = `
Object {
  "data": Object {
    "card": Object {
      "__typename": "Card",
      "created": "2020-02-01T20:19:27.909Z",
      "description": "Amet autem et.",
      "id": "NzVmNzc0NzMtYjc0Yy00YTIwLWFlMGQtNzNlMDRiNDI3Y2RlOkNhcmQ=",
      "label": "sed possimus tempore",
      "number": 99999,
      "updated": "2020-02-01T20:19:27.909Z",
    },
  },
}
`;

exports[`Server - e2e gets list of cards 1`] = `
Object {
  "data": Object {
    "cards": Object {
      "edges": Array [
        Object {
          "cursor": "Q3xjYXJkfGU5MTU3NTE3LTRkNzgtNDNkZC1hMjUxLTBkNDI2OGJkNDY1ZA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:05:43.666Z",
            "id": "ZTkxNTc1MTctNGQ3OC00M2RkLWEyNTEtMGQ0MjY4YmQ0NjVkOkNhcmQ=",
            "label": "voluptas dolores distinctio",
            "number": 12210,
            "updated": "2020-02-02T19:05:43.666Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGNjMTg4YzZhLTU1YTUtNDY0ZS04M2U2LWVlY2E3OGQzNGQ3NA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:05:38.390Z",
            "id": "Y2MxODhjNmEtNTVhNS00NjRlLTgzZTYtZWVjYTc4ZDM0ZDc0OkNhcmQ=",
            "label": "aut exercitationem laborum",
            "number": 98246,
            "updated": "2020-02-02T19:05:38.390Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDBhYmM1ODIxLTY5ZGMtNDE5Mi04MjRjLTVhM2M5MTRkZmFhZQ==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:05:33.031Z",
            "id": "MGFiYzU4MjEtNjlkYy00MTkyLTgyNGMtNWEzYzkxNGRmYWFlOkNhcmQ=",
            "label": "rerum minus iure",
            "number": 98249,
            "updated": "2020-02-02T19:05:33.031Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGRlM2I5OGViLTQ5ODktNGRmYS04MjNjLTA3MDY5ZWZiYmE3Zg==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:05:18.265Z",
            "id": "ZGUzYjk4ZWItNDk4OS00ZGZhLTgyM2MtMDcwNjllZmJiYTdmOkNhcmQ=",
            "label": "adipisci quia explicabo",
            "number": 5095,
            "updated": "2020-02-02T19:05:18.265Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGE3ZGIxY2VhLWU3ZGMtNDVkYy05NzRiLTM0OTE5MjEyY2VkMQ==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:04:38.922Z",
            "id": "YTdkYjFjZWEtZTdkYy00NWRjLTk3NGItMzQ5MTkyMTJjZWQxOkNhcmQ=",
            "label": "iste aperiam et",
            "number": 39346,
            "updated": "2020-02-02T19:04:38.922Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGRhMGU5MzA2LTk1ODctNGNmNC05YzRjLWRhZjE3MWFlOWJiNg==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:04:12.131Z",
            "id": "ZGEwZTkzMDYtOTU4Ny00Y2Y0LTljNGMtZGFmMTcxYWU5YmI2OkNhcmQ=",
            "label": "quas molestiae esse",
            "number": 29301,
            "updated": "2020-02-02T19:04:12.131Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDA1NTM3Zjc1LWRjMmUtNDc4Yy1iNzJlLWYyNmM2NzhlYTM5Nw==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:04:05.010Z",
            "id": "MDU1MzdmNzUtZGMyZS00NzhjLWI3MmUtZjI2YzY3OGVhMzk3OkNhcmQ=",
            "label": "est voluptate et",
            "number": 34517,
            "updated": "2020-02-02T19:04:05.010Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGZhOGZmOTM2LTI5MzItNDU3ZS04ZjMzLTg4YjY5Mzg0YTc1MA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:57.878Z",
            "id": "ZmE4ZmY5MzYtMjkzMi00NTdlLThmMzMtODhiNjkzODRhNzUwOkNhcmQ=",
            "label": "repellat et quia",
            "number": 42478,
            "updated": "2020-02-02T19:03:57.878Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDE2NmZjNmE1LWEwMTQtNDk1MC1hNGY3LWUyMWQ4ZjlmMWEyMg==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:54.906Z",
            "id": "MTY2ZmM2YTUtYTAxNC00OTUwLWE0ZjctZTIxZDhmOWYxYTIyOkNhcmQ=",
            "label": "officiis quibusdam ipsam",
            "number": 8553,
            "updated": "2020-02-02T19:03:54.906Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDJjZTRhOGExLTdlNjAtNGYyZS05NDBiLWMzZTM3Y2M4ZjQzMg==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:53.654Z",
            "id": "MmNlNGE4YTEtN2U2MC00ZjJlLTk0MGItYzNlMzdjYzhmNDMyOkNhcmQ=",
            "label": "omnis dolorem iusto",
            "number": 80662,
            "updated": "2020-02-02T19:03:53.654Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDE3MjBjZjc0LTdkOTItNGJmYS05MTBiLTc1MGMyZWNmYzdlNw==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:38.174Z",
            "id": "MTcyMGNmNzQtN2Q5Mi00YmZhLTkxMGItNzUwYzJlY2ZjN2U3OkNhcmQ=",
            "label": "facilis necessitatibus atque",
            "number": 92561,
            "updated": "2020-02-02T19:03:38.174Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDcyYTRiN2MzLWYxNjEtNDg1OS1iMzI4LTUzYjM3OTE5MzliMA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:33.566Z",
            "id": "NzJhNGI3YzMtZjE2MS00ODU5LWIzMjgtNTNiMzc5MTkzOWIwOkNhcmQ=",
            "label": "eos sint placeat",
            "number": 33375,
            "updated": "2020-02-02T19:03:33.566Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGIzOTE3MTk5LTAyODEtNGYxMC1iMWJlLTFjYmMwZmIyYWMxNw==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:24.499Z",
            "id": "YjM5MTcxOTktMDI4MS00ZjEwLWIxYmUtMWNiYzBmYjJhYzE3OkNhcmQ=",
            "label": "deleniti et qui",
            "number": 45499,
            "updated": "2020-02-02T19:03:24.499Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGM5ZGE2MzRmLTk4NjktNGI0MS1hY2FiLThhM2IzZTRmZjQxMg==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:22.659Z",
            "id": "YzlkYTYzNGYtOTg2OS00YjQxLWFjYWItOGEzYjNlNGZmNDEyOkNhcmQ=",
            "label": "culpa vitae est",
            "number": 38482,
            "updated": "2020-02-02T19:03:22.659Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGEzYzdhYTU1LWM4ZWYtNDczMS05NDNlLTlhNTdjMjA2NTcxYQ==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:16.177Z",
            "id": "YTNjN2FhNTUtYzhlZi00NzMxLTk0M2UtOWE1N2MyMDY1NzFhOkNhcmQ=",
            "label": "quae ullam illo",
            "number": 88443,
            "updated": "2020-02-02T19:03:16.177Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDVkMWUyOTIzLTM3MDMtNDcxZi1iZTNlLWY1ZWJhNmZiNTlmYw==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:03:11.265Z",
            "id": "NWQxZTI5MjMtMzcwMy00NzFmLWJlM2UtZjVlYmE2ZmI1OWZjOkNhcmQ=",
            "label": "cumque ipsam ut",
            "number": 14665,
            "updated": "2020-02-02T19:03:11.265Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGJjYzAzMjg2LTkyM2EtNDNkNi1hYTBkLTMyODVkNTJjZDk0ZA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:02:59.693Z",
            "id": "YmNjMDMyODYtOTIzYS00M2Q2LWFhMGQtMzI4NWQ1MmNkOTRkOkNhcmQ=",
            "label": "ducimus laboriosam porro",
            "number": 68703,
            "updated": "2020-02-02T19:02:59.693Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDJmMjRmNjg4LTEzMmMtNDFjZC04NjkwLWY5MjNlMWJkYTc0MQ==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:02:51.052Z",
            "id": "MmYyNGY2ODgtMTMyYy00MWNkLTg2OTAtZjkyM2UxYmRhNzQxOkNhcmQ=",
            "label": "unde odit laudantium",
            "number": 88574,
            "updated": "2020-02-02T19:02:51.052Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfGQxNDFiMTRlLTFmOGItNDQwZS05YjUxLTllMTBlN2ZkYmMzZA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:02:46.467Z",
            "id": "ZDE0MWIxNGUtMWY4Yi00NDBlLTliNTEtOWUxMGU3ZmRiYzNkOkNhcmQ=",
            "label": "sed modi maxime",
            "number": 67971,
            "updated": "2020-02-02T19:02:46.467Z",
          },
        },
        Object {
          "cursor": "Q3xjYXJkfDQxYjZkM2U4LTk4NjgtNDMyMC04ZDNlLWJlZTAzMDU1Y2ZmOA==",
          "node": Object {
            "__typename": "Card",
            "created": "2020-02-02T19:02:22.508Z",
            "id": "NDFiNmQzZTgtOTg2OC00MzIwLThkM2UtYmVlMDMwNTVjZmY4OkNhcmQ=",
            "label": "labore sapiente fugiat",
            "number": 36728,
            "updated": "2020-02-02T19:02:22.508Z",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "Q3xjYXJkfDQxYjZkM2U4LTk4NjgtNDMyMC04ZDNlLWJlZTAzMDU1Y2ZmOA==",
        "hasNextPage": true,
        "hasPreviousPage": false,
        "startCursor": "Q3xjYXJkfGU5MTU3NTE3LTRkNzgtNDNkZC1hMjUxLTBkNDI2OGJkNDY1ZA==",
        "totalCount": 10000,
      },
    },
  },
}
`;
